shopfloor/services/checkout.py

```diff
@@ -1405,13 +1405,14 @@ def done(self, picking_id, confirmation=False):
                 )
         lines_done = self._lines_checkout_done(picking)
         dest_location = picking.location_dest_id
-        child_locations = self.env["stock.location"].search(
-            [("id", "child_of", dest_location.id), ("usage", "!=", "view")]
-        )
-        if len(child_locations) > 0 and child_locations != dest_location:
-            return self._response_for_select_child_location(
-                picking,
+        if self.work.menu.ask_for_leaf_destination_location:
+            child_locations = self.env["stock.location"].search(
+                [("id", "child_of", dest_location.id), ("usage", "!=", "view")]
             )
+            if len(child_locations) > 0 and child_locations != dest_location:
```

Comment:
Can we simply test if `location_dest_id.usage == "view"` then we always need to choose a destination location?

Reply:
Odoo std allows to put internal locations inside internal ones, so checking only usage == `view` is not enough.
In such case we would need such option to decide what behavior we want for each Shopfloor menu.

Now I'm not sure about the current implementation of this feature, see my comment above.

Reply:
Following remarks that we are processing multiple moves, use this condition:
`len(lines_done.location_dest_id) != 1 or lines_done.location_dest_id.usage == "view"`